app/scripts/metamask-controller.js

```diff
@@ -2571,6 +2572,38 @@ export default class MetamaskController extends EventEmitter {
     if (this.onboardingController.state.completedOnboarding) {
       this.postOnboardingInitialization();
     }
+
+    this.getInfuraFeatureFlags();
+  }
+
+  // Provides a method for getting feature flags for the multichain
+  // initial rollout, such that we can remotely limit polling and/or
+  // pause polling of certain chains hosted by Infura
+  getInfuraFeatureFlags() {
+    console.log('getInfuraFeatureFlags!');
+    fetchWithCache({
+      url: 'https://swap.api.cx.metamask.io/featureFlags',
+      cacheRefreshTime: MINUTE * 20,
+    })
+      .then((response) => {
+        console.log('getInfuraFeatureFlags response:', response);
+        const { pollInterval, pausedChainIds } = response;
+        // Polling interval is provided in seconds
+        if (pollInterval > 0) {
+          this.tokenBalancesController.setIntervalLength(pollInterval * SECOND);
```

Comment:
Will require https://github.com/MetaMask/metamask-extension/pull/28472 before `this.tokenBalancesController` is defined

Reply:
PR was merged and I've rebased this PR.